### PR TITLE
Conditionally support traffic from headless agents

### PR DESCRIPTION
**Pull request by rdk08**

### CODE REVIEW

File: lib/plausible_web/controllers/api/external_controller.ex [hunk 1]
```diff
@@ -167,8 +167,13 @@ defmodule PlausibleWeb.Api.ExternalController do
 
   defp is_bot?(%UAInspector.Result.Bot{}), do: true
 
-  defp is_bot?(%UAInspector.Result{client: %UAInspector.Result.Client{name: "Headless Chrome"}}),
-    do: true
+  defp is_bot?(%UAInspector.Result{client: %UAInspector.Result.Client{name: "Headless Chrome"}}) do
+    if Application.get_env(:plausible, :system_environment) == "staging" do
```
Comment (jacooobi) [inline lib/plausible_web/controllers/api/external_controller.ex]: I think if we want to support release candidate environment we'd need to check for it as well.

Comment (jacooobi) [reply lib/plausible_web/controllers/api/external_controller.ex]: Or do we want to treat both as "staging" ? I was thinking about simply reusing the environment devops defines for their own use but I'm quite sure they set different values for staging and release candidate

Comment (rdk08) [reply lib/plausible_web/controllers/api/external_controller.ex]: Yes: https://github.com/Recruitee/k8-infrastructure/blob/master/overlays/rc/apps/careers-analytics/env.txt#L7

**Review by jacooobi — APPROVED** [review]

File: lib/plausible_web/controllers/api/external_controller.ex [hunk 2]
```diff
@@ -167,8 +167,13 @@ defmodule PlausibleWeb.Api.ExternalController do
...
   defp is_bot?(%UAInspector.Result.Bot{}), do: true
 
-  defp is_bot?(%UAInspector.Result{client: %UAInspector.Result.Client{name: "Headless Chrome"}}),
-    do: true
+  defp is_bot?(%UAInspector.Result{client: %UAInspector.Result.Client{name: "Headless Chrome"}}) do
+    if Application.get_env(:plausible, :system_environment) in ["staging", "rc"] do
+      false
+    else
+      true
+    end
```
Comment (jacooobi) [inline lib/plausible_web/controllers/api/external_controller.ex]: You can also keep it shorter by avoiding explicit return of false/true.
```
defp is_bot?(%UAInspector.Result{client: %UAInspector.Result.Client{name: "Headless Chrome"}}), do:
  Application.get_env(:plausible, :system_environment) not in ["staging", "rc"]
```

**Pull request merged**